pages/galleri/index.tsx

```diff
@@ -0,0 +1,228 @@
+import Layout, { Metadata } from '../../components/Layout';
+import styled from 'styled-components';
+import { useCallback, useEffect, useState } from 'react';
+import Image from 'next/image';
+import { itdageneBlue } from '../../utils/colors';
+
+const images = [
+  'DSC_1043.jpg',
+  'DSC_1003.jpg',
+  'DSC_1053.jpg',
+  'DSC_1105.jpg',
+  'DSC_1128.jpg',
+  'DSC_1137.jpg',
+  'DSC_1141.jpg',
+  'DSC_1145.jpg',
+];
+
+const GalleryWrap = styled('div')`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 10px;
+  align-items: center;
+  justify-content: center;
+  max-width: 1120px;
+  margin: 0 auto 60px;
+`;
+
+const SliderWrap = styled('div')`
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  z-index: 999;
+  background-color: rgba(0, 0, 0, 0.8);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  @media only screen and (max-width: 767px) {
+    display: none;
+  }
+`;
+
+const Single = styled('div')`
+  width: 350px;
+  cursor: pointer;
+`;
+
+const SingleImg = styled(Image)`
+  max-width: 100%;
+  transform: scale(1);
+  transition: transform 0.2s ease-in-out;
+
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+
+  &:hover {
+    position: relative;
+    transform: scale(1.11);
+    z-index: 99;
+    transition: transform 0.2s ease-in-out;
+  }
+`;
+
+const FullScreenImageDiv = styled('div')`
+  width: 850px;
+  cursor: pointer;
+
+  width: cals(100%-40px);
+  height: cals(100%-40px);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const FullScreenImage = styled(Image)`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+
+  max-width: 100%;
+  max-height: 100%;
+  pointer-events: none;
+  -webkit-user-select: none;
+  -ms-user-select: none;
+  user-select: none;
+`;
+
+const Title = styled('h1')`
+  font-weight: bold;
+  font-smoothing: antialiased;
+  font-size: 3rem;
+  margin-bottom: 1rem;
+`;
+
+export default function Test() {
+  const [slideNumber, setSlideNumber] = useState(0);
+  const [openModal, setOpenModal] = useState(false);
+
+  const handleOpenModal = (index: number) => {
+    setSlideNumber(index);
+    setOpenModal(true);
+  };
+
+  // Close Modal
+  const handleCloseModal = () => {
+    setOpenModal(false);
+  };
+
+  // Previous Image
+  const prevSlide = () => {
+    slideNumber === 0
+      ? setSlideNumber(images.length - 1)
+      : setSlideNumber(slideNumber - 1);
```

Comment:
A cool, but maybe more complicated way to do this is:

```ts
(images.length - 1 + slideNumber -1) % images.length -1
```

If you ever want to save precious CPU cycles :))

Reply:
It works but I am struggling to understand how. Could you explain @LudvigHz